project/src/ui/FileDialog.cpp

```diff
@@ -9,6 +9,11 @@
 
 namespace lime {
 
+	template<typename T>
+	static bool is_empty (const T* str) {
+		return str[0] == 0;
+	}
+
```

Comment:
Wouldn't `std::wcslen()` did pretty much the same thing?

Reply:
Never mind, I get it.

Reply:
Ok, I apologise for my previous useless comments but you still probably check if `str` isn't `nullptr` before dereferencing it.

Reply:
I moved the null check into the function in the latest commit.